src/RA_Core.cpp

```diff
@@ -1005,14 +1003,25 @@ API void CCONV _RA_InvokeDialog(LPARAM nID)
             break;
 
         case IDM_RA_FILES_LOGOUT:
-            RAUsers::LocalUser().Clear();
-            g_PopupWindows.Clear();
-            ra::services::ServiceLocator::Get<ra::services::IConfiguration>().Save();
-            _RA_UpdateAppTitle();
+        {
+            ra::api::Logout::Request request;
+            auto response = request.Call();
+            if (response.Succeeded())
+            {
+                RAUsers::LocalUser().Clear();
+                g_PopupWindows.Clear();
+                ra::services::ServiceLocator::Get<ra::services::IConfiguration>().Save();
+                _RA_UpdateAppTitle();
+                _RA_RebuildMenu();
 
-            ra::ui::viewmodels::MessageBoxViewModel::ShowInfoMessage(L"You are now logged out.");
-            _RA_RebuildMenu();
+                ra::ui::viewmodels::MessageBoxViewModel::ShowInfoMessage(L"You are now logged out.");
+            }
+            else
+            {
+                ra::ui::viewmodels::MessageBoxViewModel::ShowErrorMessage(L"Logout failed", ra::Widen(response.ErrorMessage));
+            }
```

Comment:
This is a pretty straight-forward example. Include the header for the API, define a request object and populate the parameters (Logout doesn't have any), then make the call. Check for success and proceed or report the error. The Login calls below are better examples.